src/modules/UI/Wallets/action.js

```diff
@@ -74,54 +73,57 @@ export const upsertWallet = (wallet: AbcCurrencyWallet) => (dispatch: any, getSt
   })
 }
 
-// setEnabledTokens is specifically for enabling them *within the GUI*, not within the core
-export const setEnabledTokens = (walletId: string, enabledTokens: any) => (dispatch: any, getState: any) => {
+// adds to core and enables in core
+export const addCustomToken = (walletId: string, tokenObj: any) => (dispatch: any, getState: any) => {
+  const state = getState()
+  const wallet = CORE_SELECTORS.getWallet(state, walletId)
+  WALLET_API.addCoreCustomToken(wallet, tokenObj)
+  .then(() => {
+    return
```

Comment:
is this necessary?

Reply:
It shouldn't be. There's an implied `return void` if you don't .then() the Promise